evergreen.yml

```diff
@@ -1582,6 +1582,47 @@ functions:
         permissions: public-read
         content_type: application/json
 
+  "static code analysis":
+    - command: shell.exec
+      type: test
+      params:
+        shell: bash
+        working_dir: mongosql-odbc-driver
+        script: |
+          ${prepare_shell}
+          export SEMGREP_APP_TOKEN=${semgrep_app_token}
+          echo "Running static code analysis with Semgrep..."
+
+          # Setup or use the existing virtualenv for semgrep
+          if [[ -f "venv/bin/activate" ]]; then
+              echo 'using existing virtualenv'
+              . venv/bin/activate
+          else
+              echo 'Creating new virtualenv'
+              python3 -m virtualenv venv
+              echo 'Activating new virtualenv'
+              . venv/bin/activate
+          fi
+          python3 -m pip install semgrep
+          # Confirm semgrep version
+          semgrep --version
+          set +e
+          semgrep --config p/rust --sarif --exclude "integration_test" --verbose --error --severity=ERROR --sarif-output=mongo-odbc-driver.sast.sarif > mongo-odbc-driver.sast.cmd.verbose.out 2>&1
```

Comment:
I looked at the SSDLC policy and Semgrep documentation again and we should be running all rules, not only the ones with severity ERROR. I'll make follow up tickets. We need to ignore rules like "Should use rust unsafe code" for example and find a way to tag other findings with False Positive or No fix needed.